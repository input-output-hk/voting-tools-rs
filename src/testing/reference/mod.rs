//! Reference database tests
//!
//! These tests also require a local postgres instance, but do not require pre-set data. These
//! tests will create a reference database, and run queries against it.
//!
//! These tests are also snapshot tests, meaning the easiest way to run them is with `cargo insta`,
//! which is provided by `flake.nix`

use crate::voting_power;

use init::reference_db;

mod init;

#[test]
fn simple_query() {
    let db = reference_db();
    let simple_result = voting_power(db, None, None, None).unwrap();
    insta::assert_json_snapshot!(simple_result);
}
