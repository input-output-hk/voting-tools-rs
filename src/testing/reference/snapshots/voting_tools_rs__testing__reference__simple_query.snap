---
source: src/testing/reference/mod.rs
expression: simple_result
---
[]
